src/Ztm.WebApi/AddressPools/RpcAddressGenerator.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using Ztm.Zcoin.Rpc;
+
+namespace Ztm.WebApi.AddressPools
+{
+    public sealed class RpcAddressGenerator : IAddressGenerator
+    {
+        readonly IZcoinRpcClient client;
+
+        public RpcAddressGenerator(IZcoinRpcClient client)
```

Comment:
We should accept factory instead. The reason is `IZcoinRpcClient` should not be a long running connection. It lifetime should be short and underlying connection should be handled by factory.